inject/src/main/java/io/avaje/inject/BeanScope.java

```diff
@@ -140,7 +141,51 @@ static BeanScopeBuilder builder() {
   <T> T get(Class<T> type, @Nullable String name);
 
   /**
-   * Return a single bean given the generic type and name.
+   * Return a single bean given its type and generic components.
+   *
+   * <pre>{@code
+   * Map<String, CoffeeMaker> coffeeMaker = beanScope.get(Map.class, String.class, CoffeeMaker.class);
+   *
+   * }</pre>
+   *
+   * @param base the base generic type
+   * @param type component types of the generic type
+   * @throws java.util.NoSuchElementException When no matching bean is found
+   */
+  default <T> T get(Type base, Type... types) {
+    return get(null, base, types);
+  }
+
+  /**
+   * Return a single bean given it's name, type, and generic components.
+   *
+   * <pre>{@code
+   * Heater heater = beanScope.get("electric", Set.class,  );
+   * heater.heat();
+   *
+   * }</pre>
+   *
+   * @param type an interface or bean type
+   * @param name the name qualifier of a specific bean
+   * @throws java.util.NoSuchElementException When no matching bean is found
+   */
+  default <T> T get(@Nullable String name, Type type, Type... types) {
+
```

Comment:
whitespace formatting
This looks like extra functionality unrelated to the bug fix.